### PR TITLE
feat: support ECS code server replicas (DS-68)

### PR DESCRIPTION
**Pull request by mlarose**
## Summary & Motivation

Support the hybrid ECS deployment with code servers/locations replicas.

## How I Tested These Changes

- unit test
- dev ECS cluster against local host cloud

### CODE REVIEW

File: python_modules/libraries/dagster-aws/dagster_aws/ecs/container_context.py [hunk 1]
```diff
@@ -186,6 +186,11 @@
                     is_required=False,
                     description="The ephemeral storage, in GiB, to use for the launched task.",
                 ),
+                "replicas": Field(
```
Comment (gibsondan) [inline python_modules/libraries/dagster-aws/dagster_aws/ecs/container_context.py:189]: we could set a default value of 1 here if nothing is set

**Review by gibsondan — DISMISSED** [review]
Just thinking through what this would look like if we also expanded it to k8s later - in k8s "resources" has a specific meaning that refers to CPU/memory/etc. and # of replicas goes in a different field. What do you think of a top-level "server_replicas" key instead of nestling it under "server_resources", for that reason? That way the docs can be more consistent at least 

Not super strongly held though, I can probably be convinced pretty easily that this is fine

**Comment by mlarose** [comment]
> Just thinking through what this would look like if we also expanded it to k8s later - in k8s "resources" has a specific meaning that refers to CPU/memory/etc. and # of replicas goes in a different field. What do you think of a top-level "server_replicas" key instead of nestling it under "server_resources", for that reason? That way the docs can be more consistent at least
> 
> Not super strongly held though, I can probably be convinced pretty easily that this is fine

The reason I nested it under server_resources is that it's being presented as ['code location resources' vs 'run resources'](https://github.com/dagster-io/dagster/pull/19711/files#diff-6e164b1724c314aa0d924159c138dad1084801dcff3fb42276a79192fd1b8cddL41-L46
). So, I thought it less confusing to keep it grouped.

Now, I can't really imagine a reason why replicas would exist for a run, so perhaps that confusion wouldn't happen, so looking at the [kubernetes configuration reference](https://github.com/dagster-io/dagster/blob/master/docs/content/dagster-cloud/deployment/agents/kubernetes/configuration-reference.mdx#L75-L92), there is still that separation between code server and run, though with entirely different names.

Thus I think it's fine as is but open to suggestion.

File: docs/content/dagster-cloud/deployment/agents/amazon-ecs/configuration-reference.mdx [hunk 2]
```diff
@@ -41,6 +41,7 @@ locations:
         server_resources: # Resources for code servers launched by the agent for this location
           cpu: 256
           memory: 512
+          replicas_count: 1
```
Comment (gibsondan) [inline docs/content/dagster-cloud/deployment/agents/amazon-ecs/configuration-reference.mdx:44]: => replica_count

**Review by gibsondan — APPROVED** [review]
drop the "s", then this looks good to me

**Pull request merged**